Dynatrace/dynatrace-operator
# Add auth token rotation for the activeGate

# Description
This change introduces the activeGate auth token rotation support. It adds the functionality that auth tokens are created with an expiration Time (60 days) and also it adds the ability to automatically rotate the tokens if they are older than 30 days.
Additionally I introduced a mechanism to restart the activeGate if the content of the auth token secret (new Auth Token) changes, by adding the auth token content to the activeGate hash annotation.


## How can this be tested?
1. Create an API Token with the correct permission `activeGateTokenManagement.create`
2. Create a Dynakube that also creates an activeGate and add `feature.dynatrace.com/enable-activegate-authtoken: "true"` to it
3. check if the activeGate gets the correct secret mounted and check the logs of the activeGate if it uses the created auth Token
4. Delete the auth-token secret in the operator namespace and check if the activeGate is restarted, when the new auth token is created
5. Wait for 30 days (or modify it when testing it and lower the rotation interval) and check if the auth token gets automatically deleted and recreated again. Additionally the activeGate pod should be restarted and get the new auth Token mounted


## Checklist
